model/model.pb.go

```diff
@@ -597,6 +597,22 @@ func (*Batch) ProtoMessage()    {}
 func (*Batch) Descriptor() ([]byte, []int) {
 	return fileDescriptor_4c16552f9fdb66d8, []int{6}
 }
+
+func (m *Batch)ConvertToTraces()(*Trace){
```

Comment:
this is auto-generated file, cannot change it manually.  Also, I am not sure why you need it - batch is just a collection of spans with a shared Process, and the HTTP handler already has code to translate a list of spans into  UI-compatible output.

Reply:
the `fromDomain` function that is used by `convertModelToUI` tries to deref Span.Process (I get a nil pointer derefencing error) : https://github.com/jaegertracing/jaeger/blob/d18b5f91e288d8af37a517ca274c7651f47c83cc/model/converter/json/from_domain.go#L60 and doesn't consider a shared common Process, would it be better to modify the `fromDomain` function instead?

Reply:
you need to prepare data for that function by denormalizing Batch.Process -> Span.Process

Reply:
That is what im trying to do in this function by copying m.Process into v.process

Reply:
I know, but (a) you cannot do this in this file, and (b) that's not all you're doing, L607-610 don't make sense, it's already being done on the GET /v1/trace code path.

Reply:
Ahhh, that makes sense, thanks! I'll revert back with the changes :)

Reply:
Yikes! I forgot models.pb is a file generated by protoc from grpc definitions...